aiida/restapi/translator/node.py

```diff
@@ -146,68 +161,181 @@ def _get_content(self):
+            # Get all attrs except those contained in nalist
+            elif self._alist is None and self._nalist is not None:
+                attrs = {}
+                for key in n.get_attrs().keys():
+                    if key not in self._nalist:
+                        attrs[key] = n.get_attr(key)
+                data = {self._content_type: attrs}
+            # Get all attrs contained in alist
+            elif self._alist is not None and self._nalist is None:
+                attrs = {}
+                for key in n.get_attrs().keys():
+                    if key in self._alist:
+                        attrs[key] = n.get_attr(key)
+                data = {self._content_type: attrs}
+            else:
+                raise RestValidationError("you cannot specify both alist "
+                                          "and nalist")
+        # content/extras
+        elif self._content_type == "extras":
+
+            # Get all extras if nelist and elist are both None
+            if self._elist is None and self._nelist is None:
+                data = {self._content_type: n.get_extras()}
+
+            # Get all extras except those contained in nelist
+            elif self._elist is None and self._nelist is not None:
+                extras = {}
+                for key in n.get_extras().keys():
+                    if key not in self._nelist:
+                        extras[key] = n.get_extra(key)
+                data = {self._content_type: extras}
+
+            # Get all extras contained in elist
+            elif self._elist is not None and self._nelist is None:
+                extras = {}
+                for key in n.get_extras().keys():
+                    if key in self._elist:
+                        extras[key] = n.get_extra(key)
+                data = {self._content_type: extras}
+
             else:
-                raise ValidationError("invalid content type")
-                # Default case
+                raise RestValidationError("you cannot specify both elist "
+                                          "and nelist")
+
+        # Data needed for visualization appropriately serialized (this
+        # actually works only for data derived classes)
+        # TODO refactor the code so to have this option only in data and
+        # derived classes
+        elif self._content_type == 'visualization':
+            # In this we do not return a dictionary but just an object and
+            # the dictionary format is set by get_visualization_data
+            data = {self._content_type: self.get_visualization_data(n)}
+
         else:
-            pass
+            raise ValidationError("invalid content type")
 
         return data
 
+    def _get_subclasses(self, parent=None, parent_class=None, recursive=True):
+        """
+         Import all submodules of the package containing the present class,
+         including subpackages recursively, if specified.
```

Comment:
The idea is to know what is the downest subclass of which the node is still an instance.

Say, you ask for "/data/768ba" and the node found is actually a KpointsData, you want to be able to use endpoints/methods of the resource "kpoints"  . This is relevant for visualization because very often you query generically for a node and you want to be able to apply the correct visualization plugin.